Android: App redesign with multi-theme system

t895 opened the pull request:
**Top-level look at changes**
+ New color mappings
+ New theme switcher with 5 base themes,
+ Several unused icons have been removed along with style declarations in favor of global style settings wherever possible.

### Dark Theme Breakdown
<details>
  <summary>Click to expand</summary>

**AlertDialog**
- Old
![old_dialog](https://user-images.githubusercontent.com/14132249/173167372-edd9912e-828d-41e1-9461-0ab29a424acc.png)
- New
![Alert](https://user-images.githubusercontent.com/14132249/179430820-6d595e04-acb7-4761-af0d-20eeb6a8c50c.png)

**Home Screen**
- Old
![old_home](https://user-images.githubusercontent.com/14132249/173167435-bea6c7c7-306a-4254-aed8-397177a9e8f2.png)
- New
![Home](https://user-images.githubusercontent.com/14132249/179430823-3e41fe23-306a-4f21-bb19-45b50fd8aaa6.png)

**Progress Bar**
- Old
![old_progressbar](https://user-images.githubusercontent.com/14132249/173167515-d47f2092-b0f5-4e5d-9a78-b4405c6f90d1.png)
- New
![progressbar](https://user-images.githubusercontent.com/14132249/179430829-a33e399b-b978-4e7d-bd15-b3cae1af7673.png)

**Settings Menu**
- Old
![old_settings](https://user-images.githubusercontent.com/14132249/173167534-fc141416-9981-4e2c-9ad3-a0a13957b277.png)
- New
![settings](https://user-images.githubusercontent.com/14132249/173204335-c31ab482-e96f-424d-ba70-34502c10a75e.png)

Note - The action bar now collapses on every settings page when scrolling

**Radio Buttons**
- Old
![old_radio](https://user-images.githubusercontent.com/14132249/173167575-39200b28-8eb0-4e07-a686-0829e8924c44.png)
- New
![radiobuttons](https://user-images.githubusercontent.com/14132249/179430836-f9522a28-9a18-4cc3-8fed-44f0dbc58c60.png)

**Sliders**
- Old
![old_slider](https://user-images.githubusercontent.com/14132249/173167613-af36368a-342a-4312-956b-86e04f8f92f9.png)
- New
![slider](https://user-images.githubusercontent.com/14132249/179430842-c43ca4a6-8034-456b-bda0-4968eafe0495.png)

**Buttons**
- Old
![old_buttons](https://user-images.githubusercontent.com/14132249/173167631-3799ab98-a695-411d-a0d4-a5950f796de0.png)
- New
![button](https://user-images.githubusercontent.com/14132249/173204345-8db5d2ec-f669-49bb-8173-5f1981761d31.png)

**Emulation Menu**
- Old
![old_emulation](https://user-images.githubusercontent.com/14132249/173167657-61bcb093-1567-4198-9a7a-147b86f89a61.png)
- New
![emu](https://user-images.githubusercontent.com/14132249/173204362-ec5e810a-4c1b-4dc9-a824-5dabd4270085.png)

**Game Properties**
- Old
![old_gameprops](https://user-images.githubusercontent.com/14132249/173197919-ab898598-e213-4bc5-bfa5-64bdec078b63.png)
- New
![props](https://user-images.githubusercontent.com/14132249/179430857-36853cb1-16a1-42bc-9fac-8be11ead2561.png)

**Game Details**
- Old
![old_details](https://user-images.githubusercontent.com/14132249/173197896-782f4706-2f77-4254-bca2-ff32548f212d.png)
- New
![details](https://user-images.githubusercontent.com/14132249/179430863-8e8c0cb2-ebef-4e23-a7e3-a42333206921.png)

**Edit Cheats**
- Old
![old](https://user-images.githubusercontent.com/14132249/173412045-a385891d-6b37-497a-b425-b929c09f5698.png)
- New
![new](https://user-images.githubusercontent.com/14132249/173412054-dd64aa3b-44d3-43cd-843d-19a3455d0bc5.png)

**Cheats downloading dialog**
- Old
![image](https://user-images.githubusercontent.com/14132249/173258865-deffbd67-f29a-4425-ae38-c14f0378e9a0.png)
- New
![Cheats downloading](https://user-images.githubusercontent.com/14132249/179430867-c62d6261-fca2-4be1-a415-c5de7afe94c2.png)

**Input Binding dialog**
- Old
![image](https://user-images.githubusercontent.com/14132249/175843996-71aaa658-b486-41e4-b3ac-84173362a2e7.png)
- New
![image](https://user-images.githubusercontent.com/14132249/175844030-ae7acdc6-c147-4a74-9ec5-f6ad4f357d27.png)

</details>

### Light Theme Breakdown
<details>
  <summary>Click to expand</summary>

**AlertDialog**
- Old
![usageold](https://user-images.githubusercontent.com/14132249/179434029-a4ba3923-321c-430f-8b7a-8327540b8d45.png)
- New
![usagenew](https://user-images.githubusercontent.com/14132249/179434032-be8a0765-b112-47bb-b540-e007b447024f.png)

**Home Screen**
- Old
![homeold](https://user-images.githubusercontent.com/14132249/179434046-a47246a5-1ecc-4b6b-ada4-32bef94e860e.png)
- New
![homenew](https://user-images.githubusercontent.com/14132249/179434041-774e5611-f7b9-4fdd-88b5-9fb67854edb7.png)

**Progress Bar**
- Old
![progressbarold](https://user-images.githubusercontent.com/14132249/179434060-8afbec29-b61d-4bc2-8396-41a4ab0f65b7.png)
- New
![progressbarnew](https://user-images.githubusercontent.com/14132249/179434063-c7bbfdde-8b6f-49ed-89b3-0fbdefffb23d.png)

**Settings Menu**
- Old
![settingsold](https://user-images.githubusercontent.com/14132249/179434076-034d75ff-170f-43d5-8b6b-60c76152f8ee.png)
- New
![settingsnew](https://user-images.githubusercontent.com/14132249/179434074-476dff2b-fb41-4701-a8af-8510965a3f98.png)

**Radio Buttons**
- Old
![radioold](https://user-images.githubusercontent.com/14132249/179434088-beb6feff-8ef0-47f5-b8c4-29cceb995a6f.png)
- New
![radionew](https://user-images.githubusercontent.com/14132249/179434091-25893eaf-8177-482e-a1e3-505b2beaeafa.png)

**Sliders**
- Old
![sliderold](https://user-images.githubusercontent.com/14132249/179434104-5b92e119-f006-47a6-90b2-774fe39a144a.png)
- New
![slidernew](https://user-images.githubusercontent.com/14132249/179434100-57b0a626-6090-4ed6-aa2c-83ae4301b1cf.png)

**Buttons**
- Old
![buttonsold](https://user-images.githubusercontent.com/14132249/179434109-f54aeddb-7ea3-426a-b150-9c4a6f6628c3.png)
- New
![buttonsnew](https://user-images.githubusercontent.com/14132249/179434115-d6d64810-d3ae-4a06-be36-19a0233b60ab.png)

**Emulation Menu**
- Old
Same as dark theme
- New
![emudialog](https://user-images.githubusercontent.com/14132249/179434159-45915705-3d85-4cb1-a6d0-086c1ff1a4d4.jpeg)

**Game Properties**
- Old
![propertiesold](https://user-images.githubusercontent.com/14132249/179434181-e5b0f764-8c83-4aca-83fa-b0b473cc005c.png)
- New
![propertiesnew](https://user-images.githubusercontent.com/14132249/179434179-a24f147f-5526-4585-964b-492c80e87dd8.png)

**Game Details**
- Old
![detailsold](https://user-images.githubusercontent.com/14132249/179434188-48e8bddb-4498-4c73-8694-e85dc03b8ebc.png)
- New
![detailsnew](https://user-images.githubusercontent.com/14132249/179434195-a69b757e-405c-4205-9ea8-c6f81e7e5238.png)

**Edit Cheats**
- Old
![cheatsold](https://user-images.githubusercontent.com/14132249/179434202-ff20c981-d5c5-422e-9ced-d0579add599b.png)
- New
![cheatsnew](https://user-images.githubusercontent.com/14132249/179434207-212e14e0-b937-475b-8cfa-28df05a76df3.png)

**Cheats downloading dialog**
- Old
![cheatsdownloadingold](https://user-images.githubusercontent.com/14132249/179434213-4bc9a01e-fb90-4d57-b6eb-f82c6b6362e2.png)
- New
![cheatsdownloadingnew](https://user-images.githubusercontent.com/14132249/179434216-9af898df-579b-4b51-8e49-2e3405df435d.png)

**Input Binding dialog**
- Old
![inputbindingold](https://user-images.githubusercontent.com/14132249/179434232-c03816e2-f929-4648-a038-0831f4ab8896.png)
- New
![inputbindingnew](https://user-images.githubusercontent.com/14132249/179434229-df29323c-2396-4eee-91ee-9e8c53a85dd6.png)

</details>

**Runtime theme change demo**
![demo](https://user-images.githubusercontent.com/14132249/179430878-14e7633b-0ef3-4f7d-8b95-cc3f8c6663ea.gif)

Note - This highlights how the default theme will stick to the strong background brand colors and how the new Material You pastel colors are shown.

**Home navigation animation demo**
![navDemo gif](https://user-images.githubusercontent.com/14132249/173204382-84efd4f0-eed4-4f69-acee-d2b65288c240.gif)

Material Default Theme (Conversion of Dolphin colors to Material You ones)
- Light
![materialdefaultlight](https://user-images.githubusercontent.com/14132249/179434485-039e9a81-c4df-4ea7-90eb-4b4f480400f3.png)
- Dark
![materialdefaultnight](https://user-images.githubusercontent.com/14132249/179434489-82ccac26-9ee9-4161-b75d-285d2b0be0e8.png)

Material You Theme (With purple wallpaper)
- Light
![purplelight](https://user-images.githubusercontent.com/14132249/179434404-27e73b9f-a404-4b5d-bc93-ccc6e75d432f.png)
- Dark
![MaterialYou_Home](https://user-images.githubusercontent.com/14132249/173167700-5eff2098-0722-4064-9ede-a70f469d1a0a.png)

Green Theme
- Light
![materialgreenlight](https://user-images.githubusercontent.com/14132249/179434386-5ec34838-28c4-4219-89f6-95c416a864e9.png)
- Dark
![Green_Home](https://user-images.githubusercontent.com/14132249/173167708-b9519f91-2cb7-4f30-9ac5-5561f2b2c070.png)

Pink Theme
- Light
![materialpinklight](https://user-images.githubusercontent.com/14132249/179434390-96b76b67-3516-48ad-a0e5-088876acc0b1.png)
- Dark
![pink](https://user-images.githubusercontent.com/14132249/174497115-abe9580f-9fe9-40bb-b8dc-59113f2508ef.png)

t895 commented:
@MayImilae and @JosJuice Let me know if there's anything I could improve here

Simonx22 commented:
This looks great overall! I will take a closer look later, but I noticed that you changed the button text color:

![image](https://user-images.githubusercontent.com/26326692/173197798-160d83ca-e2c5-417c-9d16-379a64e520a3.png)

IMO the text color shouldn't be dark here. This makes it less readable.

t895 commented:
I like the suggestion. I'll look into it, but this is done in part to account for a wider variety of theme colors.

![image](https://user-images.githubusercontent.com/14132249/173198503-b86d6ea0-1c7d-4b02-8da6-8c3971fbea66.png)
Here's an example with a Material You generated color. 

While white might work better for contrast in another theme, it struggles in others. If anything, I'll have to adjust the colors created for the default theme.

t895 commented:
Running Google's theme builder, we loose a bit of the original color but I'll leave it here.

PR - 
![home (Small)](https://user-images.githubusercontent.com/14132249/173199627-0b1b8840-36ac-4e8a-b146-a35a84badb40.png)



Theme Builder - 
![image](https://user-images.githubusercontent.com/14132249/173199489-5e4851f6-773b-4279-963e-59236edd1b3c.png)

t895 commented:
Just adjusted the colors to more closely follow the generated ones. The changes are reflected in the top comment.

Pokechu22 commented:
@dolphin-emu-bot rebuild

Pokechu22 commented:
@dolphin-emu-bot rebuild

MayImilae commented:
I like it! Good improvement. However, I have a request. Could we include a Blue theme to bring back a little bit of the old theming for those that want it?

OatmealDome commented:
@dolphin-emu-bot rebuild

t895 commented:
Just updated the MotionAlertDialog that handled binding controls. As far as I know, it is impossible to use the MaterialAlertDialogBuilder to make it consistent with the other AlertDialogs. So here I have brought it as close to what Google provides as possible with some manual changes. If I find a better solution in the future, I will either include it here or in a future PR.

MayImilae commented:
So to be more specific with my previous comment, it's the pastel-ness. See the below example.

![pastel](https://user-images.githubusercontent.com/6551020/176109160-be293c05-f19e-41c9-b12a-9ab83e76b5e2.jpg)

I know that's the current trend for android, but it's very different from how we have presented thus far and is just not aligned with Dolphin's aesthetic. IMO, that screen should be white with strong (not pastel) blue text.

 Though of course, that as a default with pastel as an option would be great. Or at least an option to get that kind of non-pastel look. I'm wanting something like that.

t895 commented:
Ah ok, I could make the default follow the previous colors with less of a pastel look but offer the the pastel one as an additional option.

MayImilae commented:
Light and dark theorectically, though the dark theme doesn't seem to need it the same way IMO. What do you think?

t895 commented:
I really like how the new dark theme colors look and as long as I remove the pastel color from dialogs, I would keep it as is.

t895 commented:
Here's what the new defaults look like

Light - 
![lightDemo gif](https://user-images.githubusercontent.com/14132249/177084075-5b4bd1ad-e792-4172-83eb-a6c21122d8bc.gif)

Dark -
![darkDemo gif](https://user-images.githubusercontent.com/14132249/177084085-2d1668cc-d291-4b93-82d3-1743beffdefd.gif)

t895 commented:
@dolphin-emu-bot rebuild

t895 commented:
Just found a couple more old dialog implementations

**Converting Dialog**
Old - 
![convertingold](https://user-images.githubusercontent.com/14132249/177627405-fb64139d-dc30-478f-8e7b-277459b2d1f0.png)

New - 
![convertingnew](https://user-images.githubusercontent.com/14132249/177627515-afb52032-dfc6-48d2-a0e8-b70a1abea244.png)

**Settings Loading Dialog**
Old - 
![settingsold](https://user-images.githubusercontent.com/14132249/177627494-0139d675-412e-4838-bbd7-6023d0533043.png)

New - 
![settingsnew](https://user-images.githubusercontent.com/14132249/177627481-7505b009-09a8-441a-b85b-a580b29c4c1b.png)

MayImilae commented:
I'm very happy with the updates. Material You support looks good, dark theme looks good, and the classic light theme is great for those wanting a more Dolphin-y look as well as our own publications showcasing Dolphin on Android. So design wise, LGTM.

t895 commented:
Just quickly brought this PR up-to-date again with the new addition of XLink Kai in #10625 and their contribution of InputStringSettings.

Here's what it looks like - 
![image](https://user-images.githubusercontent.com/14132249/183131130-d6bcbc85-0767-4a05-b555-a9366e77f7fe.png)

t895 commented:
After getting some much-needed polish in, I feel much more confident about this PR going up for review. 

After some discussion with @JosJuice I've also worked out a plan for working with both shared preferences and our config system. Whenever the theme value is saved to our config it is also saved to shared preferences as well. To account for config changes outside of dolphin, I have implemented a check for whenever MainActivity opens from a cold start. If there is a mismatch between our config and shared preferences, shared preferences will save the data from our config and the activity will be recreated. This allows a user to restore from a backup and we keep the benefit of using shared preferences to load the theme value before our UI is loaded.

Additionally, I have cleaned up the code used to create every alert dialog while I was going through changing AlertDialog.Builders to MaterialAlertDialogBuilders. Things are much more consistent now. So between the dialogs and colors/theme attributes, that is the bulk of this PR in lines of code. The parts concerning the theme system are very small in comparison.

Visually, this PR is the same as it has been for the past month, I just need feedback on the settings implementation.
(I'm also open to any additional requests for UI changes)

JosJuice commented on Source/Android/app/src/main/res/layout/activity_settings.xml:
```diff
@@ -1,5 +1,30 @@
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:background="?attr/colorSurface"
+            app:layout_scrollFlags="scroll|enterAlways|snap"/>
```
Is there a particular reason for having the behavior where this hides when scrolling down? Personally I find that behavior a little bit annoying with how tall phones are nowadays, but if there's some particular reason for it (like if it's standard in material design), I suppose it's okay.

t895 replied:
You can do either really. As long as the expanded app bar gets a drop shadow when scrolling, it stays within spec.

t895 replied:
I'm gonna try to get a collapsing toolbar layout to maybe improve the situation on taller phones

Example -
![image](https://user-images.githubusercontent.com/14132249/184548838-df45f22c-8c14-4b97-b25b-25e4d2694930.png)

t895 replied:
*Correction, you don't need the drop shadow to appear to stay in spec. It's not necessary anymore.

t895 replied:
Ok, I have the new app bar implemented I just need an opinion. Would you prefer the medium or large collapsing top app bar?

Medium - 
![image](https://user-images.githubusercontent.com/14132249/184756404-14389bb6-c0f8-4d82-8839-70365eafa00e.png)

Large -
![image](https://user-images.githubusercontent.com/14132249/184756364-33fca148-fdb1-47b1-a9fa-2fa54d1f0dca.png)

JosJuice replied:
Does the large one just have more blank space? What's the point of that?

t895 replied:
More blank space and a slightly larger title. It brings all of the settings options closer to the bottom for better reachability.

t895 replied:
Also it collapses and doesn't hide the top bar anymore

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsActivity.java:
```diff
@@ -211,11 +218,19 @@ public void showLoading()
-      dialog.setMessage(getString(R.string.load_settings));
-      dialog.setIndeterminate(true);
+      ProgressBar progressBar =
+              new ProgressBar(this, null, android.R.attr.progressBarStyle);
```
What does passing `android.R.attr.progressBarStyle` actually do here?

t895 replied:
I believe I made a mistake where I thought I was changing the theme from the indeterminate progress bar to the circular version. Additionally, there is a more appropriate material 3 provided component I'll use to replace this

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsActivity.java:
```diff
@@ -211,11 +218,19 @@ public void showLoading()
+      final int dpVertical = 16;
+      int pxHorizontal = ThemeHelper.dpToPx(dpHorizontal, getResources());
+      int pxVertical = ThemeHelper.dpToPx(dpVertical, getResources());
+      progressBar.setPadding(pxHorizontal, pxVertical, pxHorizontal, pxVertical);
```
Could it be worth defining a layout XML file for this instead of doing it all in code?

t895 replied:
Good point, that'll be in the next few commits

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsFragmentPresenter.java:
```diff
@@ -321,6 +322,52 @@ private void addInterfaceSettings(ArrayList<SettingsItem> sl)
+    {
+      sl.add(new SingleChoiceSetting(mContext, appTheme,
+              R.string.change_theme, 0, R.array.themeEntriesA12,
+              R.array.themeValuesA12));
```
You can probably afford not using this many line breaks here.

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/sysupdate/ui/SystemUpdateProgressBarDialogFragment.java:
```diff
@@ -28,27 +32,25 @@ public Dialog onCreateDialog(@Nullable Bundle savedInstanceState)
-    progressDialog.setOnShowListener((dialogInterface) ->
-    {
-      // By default, the ProgressDialog will immediately dismiss itself upon a button being pressed.
-      // Setting the OnClickListener again after the dialog is shown overrides this behavior.
```
Is this not needed anymore? Or why was it removed?

t895 replied:
Looks like it is still needed. This is one way you can prevent an AlertDialog from dismissing when pressing a button and replace it with custom behavior. I have reimplemented this in the latest commit. (It's slightly different, but it does the same thing)

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainActivity.java:
```diff
@@ -51,6 +52,8 @@ public final class MainActivity extends AppCompatActivity
   private TabLayout mTabLayout;
   private FloatingActionButton mFab;
 
+  private int themeId;
```
```suggestion
  private int mThemeId;
```

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainActivity.java:
```diff
@@ -355,4 +365,21 @@ public void onTabSelected(@NonNull TabLayout.Tab tab)
+
+  private void checkTheme()
+  {
+    ThemeHelper.checkThemeSetting(this);
```
Only calling this in MainActivity means that Dolphin's settings system will never be updated with new values from SharedPreferences if you don't pass through MainActivity. For instance, you could be using TvMainActivity, or you could be launching AppLinkActivity through a frontend app.

What do you think about moving the logic of updating Dolphin's settings system into DirectoryInitialization? We already have some code there for syncing Dolphin's settings system with SharedPreferences for Wii Remote settings. Then MainActivity would run `ThemeHelper.setCorrectTheme` after directory initialization finishes instead of `ThemeHelper.checkThemeSetting`.

t895 replied:
I put something together quick to move things into DirectoryInit. Let me know what you think.

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainActivity.java:
```diff
@@ -86,6 +94,8 @@ protected void onCreate(Bundle savedInstanceState)
   @Override
   protected void onResume()
   {
+    ThemeHelper.setCorrectTheme(this);
```
Why is this only needed in MainActivity? If the criteria are that the activity must support themes and must allow launching SettingsActivity, EmulationActivity and CheatsActivity would also need it.

t895 replied:
I was thinking that since the user has to return to MainActivity after changing their theme in settings, this would be the only place you would need to correct the activity's theme (This only applies to activities that are "Paused"). But now that I'm thinking about it, I should also account for when someone tries to change the theme while playing a game and returns to EmulationActivity. That *should* be the only other place I need to add that right?

JosJuice replied:
No, CheatsActivity can also launch the settings.

t895 replied:
I don't think you can access the interface settings from CheatsActivity

JosJuice replied:
Oh. Yeah, that's true now that I think about it.

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ThemeHelper.java:
```diff
@@ -0,0 +1,125 @@
+  public static final int GREEN = 3;
+  public static final int PINK = 4;
+
+  public static SharedPreferences mSharedPreferences;
```
```suggestion
  private static SharedPreferences mSharedPreferences;
```

JosJuice replied:
Preferably also have a comment somewhere in this file about the reason why we're using SharedPreferences. (That is, that Dolphin's usual settings system isn't initialized as early as we would need it to be.)

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ThemeHelper.java:
```diff
@@ -0,0 +1,125 @@
+  public static int dpToPx(float dp, Resources resources)
+  {
+    return (int) (dp * resources.getDisplayMetrics().density + 0.5f);
+  }
```
This doesn't really have anything to do with themes. If you really need it (would using layout XML files let us avoid having this function?), it probably fits better to put in a different file.

t895 commented:
@dolphin-emu-bot rebuild

MayImilae commented:
Could you please show video recordings of the collapsing top app bar? Not collapsing, medium collapsing, and large collapsing. I'm not an android user so I'm having a hard time visualizing it.

t895 commented:
> Could you please show video recordings of the collapsing top app bar? Not collapsing, medium collapsing, and large collapsing. I'm not an android user so I'm having a hard time visualizing it.

Medium - 
![medium](https://user-images.githubusercontent.com/14132249/185618783-d62bac17-96b1-4cbc-9319-6fb6c1df0f0f.gif)

Large - 
![large](https://user-images.githubusercontent.com/14132249/185618893-0a1576d5-d07d-492d-87d6-b6fbbb0f17da.gif)

JosJuice commented:
This is just my personal opinion and not my opinion as the Android maintainer, but that behavior just seems weird to me. I've never seen an app behave like that.

t895 commented:
> This is just my personal opinion and not my opinion as the Android maintainer, but that behavior just seems weird to me. I've never seen an app behave like that.

It's not super common right now. I've mostly seen it in some system Samsung apps and a music player I use called "Oto"

Example of a Oto's implementation  -
https://user-images.githubusercontent.com/14132249/185680048-c7aff49a-0957-4375-9bea-2d8ea66ab7b7.mp4

MayImilae commented:
Does the collasping top app bar start in the pulled down state ("medium" or "large" in the material spec), or does the title start aligned with the arrow ("small" in the spec) and then the user can pull the settings down and the settings will go lower on the screen if they so please?

t895 commented:
It starts in the pulled down state

MayImilae commented:
That seems really weird to me. In this setup, it starts closer to bottom (where their thumb will be) and then they can push it up further away from their thumb if they so wish to? But why would they ever wish to? IMO, it should start in the "small" state and let the user pull it down as they wish, or it should just be medium or large forever. Why have this switching functionality at all if it starts in the medium or large states?

Perhaps the music app you mentioned has really long lists that go well off screen, and the header shrinks during the act of scrolling down the list? That makes sense. However the example you gave is very short and doesn't benefit at all from this, and I can't really think of any point in Dolphin's GUI where it would benefit.

t895 commented:
Here's a better example

https://user-images.githubusercontent.com/14132249/185750816-13d38c67-ca2c-427e-8025-70b9d98cbde8.mp4

MayImilae commented:
That's a _very_ niche case, but that is definitely a spot where it provides a clear benefit. I don't think there are a lot of those, however, just because of how Dolphin's settings are distributed. Of course, we could just use small at all times instead and not need to worry about it. 

So, I'm kind of neutral on this detail. It doesn't hurt anything and it's a neat little visual flourish, however, I don't think it is going to really help Dolphin in any way either. So this is kind of a subjective thing. I'll leave that to people who care about this more than I do.

t895 commented:
Ok, I think it's pretty and has a little bit more utility so I don't see why not. But did you prefer the large or medium size?

MayImilae commented:
Personally I prefer Medium. Large has a very large gap that looks quite odd when there is only one word, IMO.

t895 commented:
I get that, I'll push that soon

t895 commented:
Just noticed a better way to implement our sliders with simplified theming.

Before -
Settings 
![image](https://user-images.githubusercontent.com/14132249/186236600-24fe4dab-52c6-45d4-a6a7-f8ddad74a1a7.png)

On-screen controls
![oldonscreen](https://user-images.githubusercontent.com/14132249/186236814-8570abaf-d9c9-4c1d-a3fe-434f24b7e991.png)

After -
Settings
![image](https://user-images.githubusercontent.com/14132249/186236938-7ac691a4-93a9-4a28-8e1e-34725054482d.png)

On-screen controls
![newonscreen](https://user-images.githubusercontent.com/14132249/186238081-986eee7e-1134-4a5e-a1ff-0cabb8835a7a.png)

t895 commented:
Realized that I forgot some sliders

Before - 
![signal-2022-08-24-123332_003](https://user-images.githubusercontent.com/14132249/186474038-32753f19-7dc5-4651-9727-6db35519988d.png)


After -
![signal-2022-08-24-123332_002](https://user-images.githubusercontent.com/14132249/186474015-66c6b1c7-3301-4f6a-9e30-5e0b12142f1b.png)

t895 commented:
I reworked all of the sliders and their layouts. Things are much more consistent now.
![signal-2022-08-24-133526_002](https://user-images.githubusercontent.com/14132249/186486208-7b0363f1-0605-4c23-adf0-a55cb8717c48.png)
![signal-2022-08-24-133526_003](https://user-images.githubusercontent.com/14132249/186486220-ccb192c0-b58f-44ce-9bb0-d137f0a26b0e.png)
![signal-2022-08-24-133526_004](https://user-images.githubusercontent.com/14132249/186486228-f0c3e41e-71ea-4b19-8f89-4adc8e50cbb3.png)

MayImilae commented:
The new slider layouts are way better. Having the value in the middle made no sense to me.

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsAdapter.java:
```diff
@@ -279,32 +276,28 @@ public void onSliderClick(SliderSetting item, int position)
-
-    seekbar.setOnSeekBarChangeListener(this);
+    Slider slider = view.findViewById(R.id.slider);
+    slider.setValueFrom(item.getMin());
```
Since you're calling setValueFrom, the value returned by getValue will vary between the value passed to setValueFrom and the value passed to setValueTo, right? If so, you should remove mSeekbarMinValue and all math associated with it, as mentioned in the comment that you've deleted.

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsActivity.java:
```diff
@@ -34,7 +38,9 @@ public final class SettingsActivity extends AppCompatActivity implements Setting
-  private ProgressDialog dialog;
+  private AlertDialog dialog;
+
+  public CollapsingToolbarLayout mToolbarLayout;
```
Instead of making this public, please provide a setter for the title and expose it through SettingsActivityView.

JosJuice commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ThreadUtil.java:
```diff
@@ -27,13 +29,13 @@ public static void runOnThreadAndShowResult(Activity activity, int progressTitle
     if (progressMessage != 0)
       progressDialog.setMessage(resources.getString(progressMessage));
-    progressDialog.setCancelable(false);
-    progressDialog.show();
```
Did the `show` call disappear here?

t895 commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsFragment.java:
```diff
@@ -127,7 +127,7 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
     if (titles.containsKey(menuTag))
     {
-      ((SettingsActivity) getActivity()).mToolbarLayout.setTitle(getString(titles.get(menuTag)));
+      ((SettingsActivity) getActivity()).setToolbarTitle(getString(titles.get(menuTag)));
```
Ah ok

t895 commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ThreadUtil.java:
```diff
@@ -32,7 +32,7 @@ public static void runOnThreadAndShowResult(Activity activity, int progressTitle
             .setTitle(progressTitle)
             .setCancelable(false)
-            .create();
+            .show();
```
I know what comment you're referencing. Let me check.

t895 replied:
You were right. Now I just create the dialog, set text, and then show.

JosJuice commented:
The code looks good, but there's one thing that bothers me about the design: The menu in EmulationActivity looks really plain now that its background is just white. Is there some reason why the old blue background isn't appropriate for this kind of theme? If so, do you think there's anything else we could do to make it look less plain? (Giving it the same background color as dialogs, maybe?)

t895 commented:
I'll put together a sample with the dialog colors

t895 commented:
The reason I changed the Emulation Fragment menu was because surfaces should have the most muted colors and the primary color should show itself more prominently through widgets and dialogs that appear on top of these surfaces. But here is what both options look like. Any thoughts on this, @MayImilae?

Current - 
Light
![image](https://user-images.githubusercontent.com/14132249/187342096-ace5ed20-478c-46fb-9d86-d8cc047da72c.png)

Dark
![image](https://user-images.githubusercontent.com/14132249/187342022-1ff9ddfa-d8eb-4b45-bdde-1f6a06ec77f2.png)


Suggestion - 
Light
![light](https://user-images.githubusercontent.com/14132249/187342359-04965054-0997-425e-a093-60d92402ea56.png)

Dark
![dark](https://user-images.githubusercontent.com/14132249/187342368-f0259f67-84fa-40ad-8215-f3d074d06569.png)

t895 commented:
Additionally, I used the code from Google's MaterialAlertDialogBuilder on our MotionAlertDialog so it could be even closer to the real thing. Hopefully I could get it to be perfect in a future PR, but for now this is the best we've got. The only issue is slightly different colors than MaterialAlertDialogs when using the Material You theme.

This is what they look like in the Material Default theme.

MotionAlertDialog -
![image](https://user-images.githubusercontent.com/14132249/187343884-1ff581a0-a5f4-42de-a71c-fc55efb0cc92.png)

MaterialAlertDialog -
![image](https://user-images.githubusercontent.com/14132249/187343941-9f3033bd-be5c-4eba-8352-e1646a9021ce.png)

MayImilae commented:
> The reason I changed the Emulation Fragment menu was because surfaces should have the most muted colors and the primary color should show itself more prominently through widgets and dialogs that appear on top of these surfaces. But here is what both options look like. Any thoughts on this, @MayImilae?

Personally I prefer the current images you provided over the suggestion, as I don't super care for the pastely look that Material You is going for. I don't think it will last very long (especially since it still doesn't let you just select the colours, ugh).

But, if that is what you want and it is closer to what other Android apps are doing with material you, I will not oppose it. I asked for the classic theme specifically to give an alternative from Android's current look, so there's a refuge from the pastel storm.

t895 commented:
I could still provide the option to remove the pastel look. But that would mean we would use either the look I have currently for the default theme or make an additional case with a look closer to the original. It starts to fragment things a bit though.

JosJuice commented:
> The reason I changed the Emulation Fragment menu was because surfaces should have the most muted colors and the primary color should show itself more prominently through widgets and dialogs that appear on top of these surfaces.

Well, it is kind of like a dialog in a sense. Not that it's exactly like a normal dialog, but it's something that gets shown on top of the normal surface, for a limited period of time and on a limited portion of the screen.

But if May thinks plain white looks fine, I'm going to trust her design chops over mine. Because from experience, she tends to make better choices for visual design than me :)

> But that would mean we would use either the look I have currently for the default theme

Sounds reasonable.

t895 commented:
Noted, I'll keep the plain look then. Otherwise, everything is done.
